File: LC1275.cpp
class Solution {
public:
    string tictactoe(vector<vector<int>>& moves) {
        
        int n = 3;
        
        if(moves.size() < 5)
            return "Pending";

        vector <int> r (n,0), c(n,0); 
        int d1 = 0, d2 = 0, curr = 1; // 1 -> A, -1 -> B

        for(auto i : moves){
            r[i[0]] += curr;
            c[i[1]] += curr;

            d1 = i[0] == i[1] ? d1 + curr : d1;
            d2 = i[0] + i[1] == n-1 ? d2 + curr : d2;

            if(abs(r[i[0]]) == n || abs(c[i[1]]) == n || (abs(d1) == n) || abs(d2) == n)
                return curr == 1? "A" : "B";

            curr *= -1;
        }
        
        return moves.size() < 9 ? "Pending" : "Draw";
    }
};

File: LC1275.py
class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        n = 3

        r = [0]*n
        c = [0]*n

        d1 = 0; d2 = 0; curr = 1

        for i in moves:
            r[i[0]] += curr
            c[i[1]] += curr

            d1 = d1 + curr if i[0] == i[1] else d1
            d2 = d2 + curr if i[0] + i[1] == n-1 else d2

            if abs(r[i[0]]) == n or abs(c[i[1]]) == n or abs(d1) == n or abs(d2) == n:
                return "A" if curr == 1 else "B"
            
            curr *= -1

        return "Pending" if len(moves) < 9 else "Draw"
